src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private T [] arr;
+    private int size;
+    private int currentCapacity = DEFAULT_CAPACITY;
+
+    public ArrayList() {
+        arr = (T[]) new Object [DEFAULT_CAPACITY];
+    }
+
+    public void increaseCapacity() {
+        T[] old = arr;
+        currentCapacity = currentCapacity + (currentCapacity >> 1);
+        arr = (T[]) new Object[currentCapacity];
+        for (int i = 0; i < size(); i++) {
+            arr[i] = old[i];
+        }
```

Comment:
Use [System.arraycopy()](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#arraycopy-java.lang.Object-int-java.lang.Object-int-int-) to move your array elements.
https://mate-academy.github.io/jv-program-common-mistakes/java-core/collections/array-list.html#dont-create-redundant-variables